crates/zed/src/languages/haskell.rs

```diff
@@ -0,0 +1,55 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use language::{LanguageServerName, LspAdapter, LspAdapterDelegate};
+use lsp::LanguageServerBinary;
+use std::{any::Any, path::PathBuf};
+
+pub struct HaskellLanguageServer;
+
+#[async_trait]
+impl LspAdapter for HaskellLanguageServer {
+    fn name(&self) -> LanguageServerName {
+        LanguageServerName("hls".into())
+    }
+
+    fn short_name(&self) -> &'static str {
+        "hls"
+    }
+
+    async fn fetch_latest_server_version(
+        &self,
+        _: &dyn LspAdapterDelegate,
+    ) -> Result<Box<dyn 'static + Any + Send>> {
+        Ok(Box::new(()))
+    }
+
+    async fn fetch_server_binary(
+        &self,
+        _version: Box<dyn 'static + Send + Any>,
+        _container_dir: PathBuf,
+        _: &dyn LspAdapterDelegate,
+    ) -> Result<LanguageServerBinary> {
+        Err(anyhow!(
+            "hls (haskell language server) must be installed via ghcup"
+        ))
+    }
+
+    async fn cached_server_binary(
+        &self,
+        _: PathBuf,
+        _: &dyn LspAdapterDelegate,
+    ) -> Option<LanguageServerBinary> {
+        Some(LanguageServerBinary {
+            path: "haskell-language-server-wrapper".into(),
```

Comment:
Is this still the correct binary to be invoking, now that we're sourcing it from the `PATH`?

Reply:
Yep, HLS ships with a few separate binaries for compatibility with different versions of GHC, the wrapper automatically selects the correct version.

Reply:
perfect, thank you!